go-controller/pkg/ovn/kubevirt.go

```diff
@@ -110,18 +121,263 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
 	return libovsdbops.DeleteDHCPOptionsWithPredicate(oc.nbClient, predicate)
 }
 
-func (oc *DefaultNetworkController) kubevirtCleanUp(pod *corev1.Pod) error {
-	if kubevirt.IsPodLiveMigratable(pod) {
-		isMigratedSourcePodStale, err := kubevirt.IsMigratedSourcePodStale(oc.watchFactory, pod)
+func (oc *DefaultNetworkController) deleteStaticRoutesForMigratedPod(pod *kapi.Pod) error {
+	vm := kubevirt.ExtractVMNameFromPod(pod)
+	routePredicate := func(item *nbdb.LogicalRouterStaticRoute) bool {
+		return kubevirt.ExternalIDsContainsVM(item.ExternalIDs, vm)
+	}
+	if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, routePredicate); err != nil {
+		return fmt.Errorf("failed deleting pod routing when deleting the LR static routes: %v", err)
+	}
+	policyPredicate := func(item *nbdb.LogicalRouterPolicy) bool {
+		return kubevirt.ExternalIDsContainsVM(item.ExternalIDs, vm)
+	}
+	if err := libovsdbops.DeleteLogicalRouterPoliciesWithPredicate(oc.nbClient, types.OVNClusterRouter, policyPredicate); err != nil {
+		return fmt.Errorf("failed deleting pod routing when deleting the LR policies: %v", err)
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) cleanUpForVM(pod *corev1.Pod) error {
+	isMigratedSourcePodStale, err := kubevirt.IsMigratedSourcePodStale(oc.watchFactory, pod)
+	if err != nil {
+		return fmt.Errorf("failed cleaning up VM when checking if pod is leftover: %v", err)
+	}
+	// Everything has already being cleand up since this is and old migration
+	// pod
+	if isMigratedSourcePodStale {
+		return nil
+	}
+	// This pod is not part of ip migration so we don't need to clean up
+	if !kubevirt.IsPodLiveMigratable(pod) {
+		return nil
+	}
+	if err := oc.deleteDHCPOptions(pod); err != nil {
+		return err
+	}
+	if err := oc.deleteStaticRoutesForMigratedPod(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureLocalZonePodAddressesToNodeRoute(pod *kapi.Pod) error {
```

Comment:
Can you please explain in the documentation of this method how this routing works in context of the routing that is already in place? This depends on other policies/routes already in place and it is hard to follow.